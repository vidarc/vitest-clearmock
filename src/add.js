export const add = (a, b) => a + b;

export const makeAnObject = () => ({
  value: 1,
  name: "hello",
  nested: {
    value: 2,
  },
  array: [1, 2, 3],
});
