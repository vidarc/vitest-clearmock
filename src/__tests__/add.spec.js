import fc from "fast-check";
import { describe, expect, test } from "vitest";

import { add, makeAnObject } from "../add.js";

describe("testing add function", () => {
  test("adds correctly", () => {
    fc.assert(
      fc.property(fc.integer(), fc.integer(), (a, b) => {
        const result = add(a, b);
        const expected = a + b;
        expect(result).toEqual(expected);
      }),
      { verbose: true }
    );
  });

  test("a failing test", () => {
    const result = add(2, 2);

    expect(result).toStrictEqual(4);
  });

  describe("when a test is nested", () => {
    test("it should pass", () => {
      expect(true).toBe(true);
    });

    test("it should fail", () => {
      expect(true).toBe(true);
    });

    test("snapshot", () => {
      expect(makeAnObject()).toMatchSnapshot();
    });

    test("inline snapshot", () => {
      expect(makeAnObject()).toMatchInlineSnapshot();
    });
  });
});
