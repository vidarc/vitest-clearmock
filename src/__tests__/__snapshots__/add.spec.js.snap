// Vitest Snapshot v1

exports[`testing add function > when a test is nested > snapshot 1`] = `
{
  "array": [
    1,
    2,
    3,
  ],
  "name": "hello",
  "nested": {
    "value": 2,
  },
  "value": 1,
}
`;
